container/deps/vllm/install_vllm.sh

```
@@ -147,22 +147,45 @@ echo "\n=== Installing vLLM-Omni ==="
 if [ -n "$VLLM_OMNI_REF" ] && [ "$ARCH" = "amd64" ]; then
-    # Try PyPI first, fall back to building from source
-    if uv pip install vllm-omni==${VLLM_OMNI_REF#v} 2>&1; then
-        echo "✓ vLLM-Omni ${VLLM_OMNI_REF} installed from PyPI"
-    else
-        echo "⚠ PyPI install failed, building from source..."
+    if [ "$DEVICE" = "xpu" ]; then
+        # XPU: build from source with XPU-specific env vars (cannot pip install)
+        echo "Building vLLM-Omni from source for XPU..."
+        export VLLM_OMNI_TARGET_DEVICE=xpu
+        export VLLM_WORKER_MULTIPROC_METHOD=spawn
         git clone --depth 1 --branch ${VLLM_OMNI_REF} https://github.com/vllm-project/vllm-omni.git $INSTALLATION_DIR/vllm-omni
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)